### PR TITLE
[WIP] Implementation for `dpnp.zeros()` and `dpnp.ones()`

### PR DESCRIPTION
**Pull request by akharche**
- Create DPCTL queue_memset wrapper
- Implement typer for dpnp.zeros and dpnp.ones
- Implement lowere for dpnp.zeros with memset
- Create tests

Depends on dpctl.memset implementation

### CODE REVIEW

File: numba_dppy/dpctl_iface/dpctl_capi_fn_builder.py [hunk 1]
```diff
@@ -295,3 +295,38 @@ def get_dpctl_event_delete(builder, context):
...
             func_name="DPCTLEvent_Delete",
         )
+
+    @staticmethod
+    def get_dpctl_queue_memset(builder, context):
+        """
+        get_dpctl_queue_memset(builder, context)
+        Inserts an LLVM Function for the ``DPCTLQueue_Memset``.
+
+        The ``DPCTLQueue_Memset`` function is a wrapper over ``sycl::queue``
```
Comment (diptorupd) [inline numba_dppy/dpctl_iface/dpctl_capi_fn_builder.py:305]: dpctl currently does not expose a function `DPCTLQueue_Memset` function. It needs to be first added to dpctl's `libsyclinterface` library. Can you open a PR against dpctl as well? A while back dpctl memory functions where changes to return event refs and the Memset wrapper will follow the same design (in fact the signature you have in the docstring does return an event). Therefore, remove the comment about "the DPCTLQueue_Memset does not return an event reference".

File: numba_dppy/dpctl_iface/dpctl_capi_fn_builder.py [hunk 2]
```diff
@@ -295,3 +295,38 @@ def get_dpctl_event_delete(builder, context):
...
+            context: The LLVM IR builder context.
+
+        Return: A Python object wrapping an LLVM Function for
+                ``DPCTLQueue_Memset``.
+        """
+        void_ptr_t = utils.get_llvm_type(context=context, type=types.voidptr)
+        return DpctlCAPIFnBuilder._build_dpctl_function(
+            builder,
+            return_ty=void_ptr_t,
+            # return_ty=utils.LLVMTypes.void_t,
```
Comment (diptorupd) [inline numba_dppy/dpctl_iface/dpctl_capi_fn_builder.py:323]: Remove commented out code.

File: numba_dppy/dpctl_iface/dpctl_capi_fn_builder.py (hunk 1 above)
Comment (akharche) [reply numba_dppy/dpctl_iface/dpctl_capi_fn_builder.py:305]: Yes, I made a request to dpctl for memset implementation.

Comment (diptorupd) [reply numba_dppy/dpctl_iface/dpctl_capi_fn_builder.py:305]: I meant can you open a PR providing the implementation. @1e-to can you help as you have lots of experience with libsyclinterface.

Comment (oleksandr-pavlyk) [reply numba_dppy/dpctl_iface/dpctl_capi_fn_builder.py:305]: The `DPCTLQueue_Memset` has been added in IntelPython/dpctl#812 and exposed via Python API in IntelPython/dpctl#815.

**Comment by diptorupd** [comment]
Superseded by #923

**Pull request closed without merging**